src/de.rs

```diff
@@ -399,6 +559,227 @@ impl<'de> de::Deserializer<'de> for StringDeserializer {
 /// This conversion can fail if the structure of the `Value` does not match the
 /// structure expected by `D`.
 pub fn from_value<'de, D: Deserialize<'de>>(value: &'de Value) -> Result<D, Error> {
-    let mut de = Deserializer::new(value);
-    D::deserialize(&mut de)
+    let de = Deserializer::new(value);
+    D::deserialize(& de)
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
```

Comment:
These tests use `serde::Serialize` but it's not defined above, so it should be defined here as

```suggestion
    use super::*;
    use serde::Serialize;
```

I tried this locally and all tests passed, so hopefully it's as simple as that! It was really sweet to see enums parsed effortlessly when I tried out this branch